applications/tari_console_wallet/src/automation/command_parser.rs

```diff
@@ -97,11 +98,25 @@ pub fn parse_command(command: &str) -> Result<ParsedCommand, ParseError> {
         MakeItRain => parse_make_it_rain(args)?,
         CoinSplit => parse_coin_split(args)?,
         DiscoverPeer => parse_discover_peer(args)?,
+        Whois => parse_whois(args)?,
     };
 
     Ok(ParsedCommand { command, args })
 }
 
+fn parse_whois(mut args: SplitWhitespace) -> Result<Vec<ParsedArgument>, ParseError> {
+    let mut parsed_args = Vec::new();
+
+    // public key/emoji id
+    let pubkey = args
+        .next()
+        .ok_or_else(|| ParseError::Empty("public key or emoji id".to_string()))?;
+    let pubkey = parse_emoji_id_or_public_key(pubkey).ok_or(ParseError::PublicKey)?;
+    parsed_args.push(ParsedArgument::PublicKey(pubkey));
+
+    Ok(parsed_args)
```

Comment:
(just ignore this) nit:

```suggestion
 let args = args
        .next()
        .ok_or_else(|| ParseError::Empty("public key or emoji id".to_string()))
        .and_then(|pk| parse_emoji_id_or_public_key(pk).ok_or(ParseError::PublicKey))
        .map(ParsedArgument::PublicKey)?;

    Ok(vec![args])
```

Reply:
That's awesome. Will refactor some time.